File: src/main/java/com/xavierbouclet/whiskies/api/configuration/WhiskyClientProperties.java
package com.xavierbouclet.whiskies.api.configuration;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotNull;

@Validated
@ConfigurationProperties(prefix = "client.whisky.service")
@ConstructorBinding
public class WhiskyClientProperties {

    @NotNull
    private final String url;

    public WhiskyClientProperties(String url) {
        this.url = url;
    }

    public String url() {
        return url;
    }
}

File: src/main/kotlin/com/xavierbouclet/whiskies/api/configuration/WhiskyClientProperties.kt
package com.xavierbouclet.whiskies.api.configuration

import jakarta.validation.constraints.NotNull
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "client.whisky.service")
data class WhiskyClientProperties(val url: @NotNull String)